From broonie@opensource.wolfsonmicro.com  Fri Jul 17 07:28:39 2009
From: Mark Brown <broonie@opensource.wolfsonmicro.com>
Date: Fri, 17 Jul 2009 15:06:08 +0100
Subject: Driver core: Add accessor for device platform data
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org, Mark Brown <broonie@opensource.wolfsonmicro.com>
Message-ID: <1247839568-23225-1-git-send-email-broonie@opensource.wolfsonmicro.com>


For consistency with driver data provide a dev_get_platdata() accessor
for reading the platform data from a device.

Signed-off-by: Mark Brown <broonie@opensource.wolfsonmicro.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/device.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -493,6 +493,11 @@ static inline struct device *root_device
 }
 extern void root_device_unregister(struct device *root);
 
+static inline void *dev_get_platdata(const struct device *dev)
+{
+	return dev->platform_data;
+}
+
 /*
  * Manual binding of a device to driver. See drivers/base/bus.c
  * for information on use.
